server/graphql/v2/mutation/OrderMutations.js

```diff
@@ -253,7 +253,7 @@ const orderMutations = {
       checkRemoteUserCanUseOrders(req);
 
       const decodedId = idDecode(args.order.id, IDENTIFIER_TYPES.ORDER);
-      const haveDetailsChanged = !isUndefined(args.amount) && !isUndefined(args.tier);
+      const haveDetailsChanged = !isUndefined(args.amount) || !isUndefined(args.tier);
```

Comment:
Small change here as it was making my test fail: it's valid to update the amount of an order without updating the tier